### PR TITLE
AWS: Add support for custom taints and labels

### LINKED ISSUE #4: Set custom labels and taints for AWS platform
## Description

In Lokomotive for AWS add support for custom labels and taints for worker nodes. Just like packet has.

- Custom Labels: https://github.com/kinvolk/lokomotive-kubernetes/blob/0da7e1426119a6b1539ff45a2f0adbcc7198842b/packet/flatcar-linux/kubernetes/workers/variables.tf#L37-L43
- Custom Taints: https://github.com/kinvolk/lokomotive-kubernetes/blob/0da7e1426119a6b1539ff45a2f0adbcc7198842b/packet/flatcar-linux/kubernetes/workers/variables.tf#L45-L49

## Initial task breakdown

- [ ] Add the variables like it is done in Packet.
- [ ] Consume them at the Kubelet level flags.
- [ ] Make sure to port those above changes to the Lokoctl.
- [ ] Add test to lokoctl which verify that such labels/taints have been added.
- [ ] Add docs that show how to use this feature.

## Acceptance Criteria

- [ ] User can set labels and taints from lokoctl and create cluster and verify that those have been applied succefully.

## Follow up work

- Create similar issues/work for other supported providers (Azure, Baremetal, Google Cloud, KVM Libvirt)

### PR DESCRIPTION
**Pull request by knrt10**
fixes #4 

Signed-off-by: knrt10 kautilya@kinvolk.io

### CODE REVIEW

File: pkg/platform/aws/template.go [hunk 1]
```diff
@@ -139,6 +139,14 @@ module "worker-pool-{{ $index }}" {
   os_version            = "{{ $pool.OSVersion }}"
   {{- end }}
 
+  {{- if $pool.Labels }}
+  labels                = "{{ $pool.Labels }}"
```
Comment (invidian) [inline pkg/platform/aws/template.go:143]: It's a value separated by new line, so the alignment should "reset".
```suggestion
  labels = "{{ $pool.Labels }}"
```

File: pkg/platform/aws/template.go [hunk 2]
```diff
@@ -139,6 +139,14 @@ module "worker-pool-{{ $index }}" {
...
+  {{- end}}
+
+  {{- if $pool.Taints }}
+  taints                = "{{ $pool.Taints }}"
```
Comment (invidian) [inline pkg/platform/aws/template.go:147]:
```suggestion
  taints = "{{ $pool.Taints }}"
```

**Review by invidian — COMMENTED** [review]
Just some nits, so LGTM.

I wonder, perhaps we should also add tests for that, to make sure the labels and taints are properly applied and retained.

**Comment by knrt10** [comment]
Yes, I was thinking to add tests, but was confused where to add them. Can you suggest @invidian ?

**Comment by invidian** [comment]
>Yes, I was thinking to add tests, but was confused where to add them. Can you suggest @invidian ?

Sure. I'd do the following:
- add extra worker pool with some very small instance type and spot price, so we don't pay much for it, with labels and taints set.
- Modify `test/components/kubernetes/kubelet_labels_test.go` and `test/components/kubernetes/kubelet_disruptive_test.go
` to also include AWS
- in `test/components/kubernetes` add a test to verify that the node has taint applied
- modify `test/components/kubernetes/kubelet_disruptive_test.go` to also check, that the taint has been retained during node re-registration.

File: test/components/kubernetes/kubelet_taint_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build aws
+// +build e2e
+
+package kubernetes //nolint:testpackage
```
Comment (invidian) [inline test/components/kubernetes/kubelet_taint_test.go:18]: IMO `kubernetes_test` should be fine here.

File: test/components/kubernetes/kubelet_taint_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+	for _, items := range nodes.Items {
+		nodeName := items.Name
+
+		for _, taint := range items.Spec.Taints {
+			taintKey := taint.Key
+			taintValue := taint.Value
+			taintEffect := taint.Effect
+
+			t.Run(nodeName, func(t *testing.T) {
```
Comment (invidian) [inline test/components/kubernetes/kubelet_taint_test.go:54]: I think we don't need either `t.Run` or `wait.PollImmidiate` here.

**Review by invidian — CHANGES_REQUESTED** [review]

File: ci/aws/aws-cluster.lokocfg.envsubst [hunk 5]
```diff
@@ -59,6 +59,36 @@ EOF
...
 
+  worker_pool "worker-pool-2" {
+    count         = 1
+    ssh_pubkeys   = ["$PUB_KEY"]
+    disk_size     = 30
+    instance_type = "t2.small"
+    spot_price    = "0.01"
+    labels        = "testing.io=yes,roleofnode=testing"
+    taints        = "nodeType=storage:NoSchedule"
+    lb_http_port  = 8080
```
Comment (invidian) [inline ci/aws/aws-cluster.lokocfg.envsubst:70]: Can you please add a comment, that this is a workaround for a bug (insert issue link here)?

**Review by invidian — CHANGES_REQUESTED** [review]
Just one nit, otherwise LGTM.

**Review by invidian — APPROVED** [review]
LGTM

**Review by invidian — APPROVED** [review]

**Pull request merged**